### ISSUE TITLE
only allow specific characters in labels 

### ISSUE THREAD

**Issue by bbernhard** [bbernhard/imagemonkey-core#183]
We shouldn't allow `|`, `~`, `&`, `(` and `)` in label names as those characters are part of the query syntax. 

It probably makes sense to implement #182 first, before we start with that. 

reminder for myself: after this ticket is done, remove all label suggestions that start with `~` - those were marked by @dobkeratops for deletion :)

**Comment by dobkeratops** [comment]
I'd agree with this .  the ```(..)``` is unfortunate because it's used for disambiguation in wikipedia titles- but reserving those characters for operator based queries is highly compelling.

I wondered if you could distinguish this use in the same way as a programming language parser distinguishing them in the case of function calls vs grouping (eg..   ```foo(bar)``` makes the parens a function-call wheras  ```foo|(bar&baz)``` would parse it as grouping ... but that's getting rather complicated

perhaps we could look specifically at the ambiguous words and see if we have other workarounds?

```glass``` is one  glaring example [glass (material)](https://en.wikipedia.org/wiki/Glass) - wikipedia's default use , vs [glass (drink ware)](https://en.wikipedia.org/w/index.php?title=Glass_(drinkware)&redirect=no).  would the blend work ("glass/cup") since the object is still a peice of glass?  [cup](https://en.wikipedia.org/wiki/Cup) - "It can be made of wood, plastic, glass, clay, metal, stone, china or other materials"

**Comment by bbernhard** [comment]
You are right, the brackets could indeed be helpful for clarification. Many thanks for bringing that up!

> I wondered if you could distinguish this use in the same way as a programming language parser distinguishing them in the case of function calls vs grouping (eg.. foo(bar) makes the parens a function-call wheras foo|(bar&baz) would parse it as grouping ... but that's getting rather complicated.

that's a good idea! I'll have a look, if I can add that to the parser implementation. Currently the parser is a little piece of code that I hacked together in an afternoon. I guess it might make sense at some point to replace the self-written parser with something more robust (parser generator) :)
